### DISCUSSION TITLE
w_advection

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#17311] [Q&A Modules: General] [answered]
w (z velocity) is not recognized in NSFVMomentumAdvection, INSFVMomentumPressure , FVDiffusion. What file will I need to modify in order to include w?

**Comment by GiudGiud** [comment, marked as answer]
w should be the variable, not an extra argument, for INSFVMomentumPressure and FVDiffusion. 
Could you please paste here your FVKernel syntax?

Thanks,
Guillaume

**Reply by TLWise** [reply]
Thank you for you interest in potentially helping me to solve this issue. Here is the syntax for the FVKernels:

[FVKernels]
  [mass]
    type = INSFVMassAdvection
    variable = pressure
    advected_interp_method = ${advected_interp_method}
    velocity_interp_method = ${velocity_interp_method}
    vel = 'velocity'
    pressure = pressure
    u = u
    v = v
    w = w
    mu = ${mu}
    rho = ${rho}
  []

  [u_advection]
    type = INSFVMomentumAdvection
    variable = u
    advected_quantity = 'rhou'
    vel = 'velocity'
    advected_interp_method = ${advected_interp_method}
    velocity_interp_method = ${velocity_interp_method}
    pressure = pressure
    u = u
    v = v
    w = w
    mu = ${mu}
    rho = ${rho}
  []
  
  [u_viscosity]
    type = FVDiffusion
    variable = u
    coeff = ${mu}
  []
  
  [u_pressure]
    type = INSFVMomentumPressure
    variable = u
    momentum_component = 'x'
    p = pressure
  []

  [v_advection]
    type = INSFVMomentumAdvection
    variable = v
    advected_quantity = 'rhov'
    vel = 'velocity'
    advected_interp_method = ${advected_interp_method}
    velocity_interp_method = ${velocity_interp_method}
    pressure = pressure
    u = u
    v = v
    mu = ${mu}
    rho = ${rho}
  []
  
  [v_viscosity]
    type = FVDiffusion
    variable = v
    coeff = ${mu}
  []
  
  [v_pressure]
    type = INSFVMomentumPressure
    variable = v
    momentum_component = 'y'
    p = pressure
  []

[w_advection]
  type = INSFVMomentumAdvection
  variable = w
  advected_quantity = 'rhow'
  vel = 'velocity'
  advected_interp_method = ${advected_interp_method}
  velocity_interp_method = ${velocity_interp_method}
  pressure = pressure
  u = u
  v = v
  w = w
  mu = ${mu}
  rho = ${rho}
[]

[w_viscosity]
  type = FVDiffusion
  variable = w
  coeff = ${mu}
[]

[w_pressure]
  type = INSFVMomentumPressure
  variable = w
  momentum_component = 'z'
  p = pressure
[]

[]

**Reply by GiudGiud** [reply]
It looks like you are missing w = 'w' for the v_advection kernel.
I ll check it more offline

**Reply by GiudGiud** [reply]
It runs fine for me now when moving those kernels with the fix in an example problem

**Reply by TLWise** [reply]
Wow! That is awesome! Thank you for your sage guidance on this problem.
